#14119 Missing feature "Anonymize contact" / feature to add new event responsible to anonymize user data

lenonleite opened the pull request:
- tests
- add field list type
- update lead events
- add helper anonymize
- change domain test.

<!-- ## Which branch should I use for my PR?

Assuming that:

a = current major release
b = current minor release
c = future major release

* a.x for any features and enhancements (e.g. 5.x)
* a.b for any bug fixes (e.g. 4.4, 5.1)
* c.x for any features, enhancements or bug fixes with backward compatibility breaking changes (e.g. 5.x) -->

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | 🔴🟢 <!-- Use emojis to indicate positive (green) or negative (red) for each item in the table. -->
| New feature/enhancement? (use the a.x branch)      | 🔴🟢
| Deprecations?                          | 🔴🟢
| BC breaks? (use the c.x branch)        | 🔴🟢
| Automated tests included?              | 🔴🟢 <!-- All PRs must maintain or improve code coverage -->
| Related user documentation PR URL      | mautic/user-documentation#... <!-- required for new features -->
| Related developer documentation PR URL | mautic/developer-documentation-new#... <!-- required for developer-facing changes -->
| Issue(s) addressed                     | Fixes #... <!-- prefix each issue number with "Fixes #", no need to create an issue if none exists, explain below instead -->

<!--
Additionally (see https://contribute.mautic.org/contributing-to-mautic/developer/code/pull-requests#work-on-your-pull-request):
 - Always add tests and ensure they pass.
 - Bug fixes must be submitted against the lowest maintained branch where they apply
   (lowest branches are regularly merged to upper ones so they get the fixes too.)
 - Features and deprecations must be submitted against the "4.x" branch.
-->

## Description

Feature responsible to create new event action that can anonymize data. 
It's based on issue #14119 


<!--
Please write a short README for your feature/bugfix. This will help people understand your PR and what it aims to do. If you are fixing a bug and if there is no linked issue already, please provide steps to reproduce the issue here.
-->
<!-- Remove HTML comment markup below to use the table for screenshots when relevant. -->
<!--
| Before                                 | After
| -------------------------------------- | ---
|                                        | 
-->


---
### 📋 Steps to test this PR:

<!--
This part is crucial. Take the time to write very clear, annotated and step by step test instructions, because testers may not be developers.
-->
1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Add new Contact with
    - Email
    - First Name
    - Last Name
    - Position
    - add tag 'Anonymize"
 
![mautic-issue-new-action-contact-form-1](https://github.com/user-attachments/assets/910b6f91-f530-4550-b6f8-f462ec628788)

3. Create a Segment name "_SegToAnonimyze_" with filter to add contacts with **Anonymize** tag
4. Run command `mautic:segments:update` to add contacts in segments
5. Create a Campaign based with segment SegToAnonimyze
6. Add new action with name "Anonymize User Data"
![Screenshot from 2024-10-07 14-08-16](https://github.com/user-attachments/assets/69e9fd33-6126-431d-8e6c-608a3b58e92e)

8. Fill form like:
    - Name : test
    - Execute this event : Immedialty
    - Pseudonymize data? Yes
    - Anonymize User Data from fields : Last Name
    - Delete User Data from fields : Position
    
![mautic-issue-new-action-form-edited](https://github.com/user-attachments/assets/30683fbc-14e1-4846-9b4f-199e1b49f59d)

8 - Save Campaign 
9 - Run command "mautic:campaigns:rebuild" to build campaign
10 - Run command  mautic:campaigns:trigger
11 - Come back to contact created and check:
    - Last name was change by hash
    - Position is empty

![mautic-issue-new-action-contact-form-edited](https://github.com/user-attachments/assets/f07e6f0f-3ade-498a-b808-05b20ddf3f03)

<!--
If you have any deprecations and backwards compatibility breaks, list them here along with the new alternative.
-->

lenonleite committed:
- mst-83

lenonleite committed:
- mst-83

lenonleite committed:
- mst-83

lenonleite committed:
- mst-83

lenonleite committed:
- mst-83

lenonleite committed:
- mst-83

lenonleite committed:
- mst-83
- add viewother permissions

lenonleite committed:
- mst-83
- add viewother permissions

escopecz commented:
@lenonleite please update the description describing what this does and how to test it. This will also need an update to the user docs.

lenonleite commented:
@escopecz, ok I will do it.

putzwasser committed:
FIX: Ignoring of custom option list for mapped fields (#14117)

This PR fixes #13306 and #13542.

kuzmany reviewed (approved):
Nice feature. 
I test campaign action. 
It works properly. 
Only seems option "Pseudonymize data?"  is useless.  I would remove it. 

![image](https://github.com/user-attachments/assets/d5a52677-c66f-4bec-b92c-288a78682bb5)

kuzmany reviewed (dismissed):
Nice feature. 
I test campaign action. 
It works properly. 
Only seems option "Pseudonymize data?"  is useless.  I would remove it. 

![image](https://github.com/user-attachments/assets/d5a52677-c66f-4bec-b92c-288a78682bb5)

patrykgruszka committed:
fix: [DPMMA-2957] message queue orm issue

shinde-rahul committed:
hash sensitive header data for logging

shinde-rahul committed:
Added coverage

shinde-rahul committed:
Update app/bundles/PluginBundle/Tests/Helper/oAuthHelperTest.php

shinde-rahul committed:
Added basic authentication as well.

shinde-rahul committed:
Fixed cs

escopecz committed:
Merge pull request #2391 from acquia/MAUT-11975

Maut 11975 - Fixing segment building with default timezone

escopecz committed:
Merge pull request #2388 from acquia/MAUT-11969

Maut 11969 - Email click tracking fix, PHP warning fix

escopecz committed:
Revert "delete transient test"

This reverts commit da65d89071807f1ea83cc2a4e091cae461ad11b9.

escopecz committed:
Taking changes from the community

escopecz committed:
Setting UTC timezone as it will fail if you have different timezone in your local config

escopecz committed:
More readable and extendable assertions

escopecz committed:
Execute the event immediately if the "send from" hour has passed

escopecz committed:
Resolving PHPSTAN and flakyness of this test

escopecz committed:
Use entities over mocks to make the test x-debuggable

escopecz committed:
Fixing splitting contacts into different time groups. The timestamp() is always returning UTC. So it didn't really work. The key must be a time format.

shinde-rahul committed:
Updated sanitization to replace hashed value with [REDACTED]

escopecz committed:
Updating existing tests to execute earlier now

escopecz committed:
Resolving time issue in the tests

biozshock committed:
SAMl login fix and test.

escopecz committed:
STAN fixes

escopecz committed:
It makes more sense to have `getIsScheduled` set to false for events that should execute immediately

andersonjeccel committed:
up font-size

andersonjeccel committed:
asset again

andersonjeccel committed:
improving contact results

andersonjeccel committed:
improving landing page results

andersonjeccel committed:
improving user results

andersonjeccel committed:
improving results: campaigns

andersonjeccel committed:
improving results: emails

andersonjeccel committed:
improving results: forms

andersonjeccel committed:
improving results: pages

andersonjeccel committed:
improving results: point actions

andersonjeccel committed:
improving results: point triggers

andersonjeccel committed:
improving results: reports

andersonjeccel committed:
improving results: stages

andersonjeccel committed:
improving results: roles

andersonjeccel committed:
global search styles improvement

andersonjeccel committed:
JS for gsearch

andersonjeccel committed:
final adjustments to the template

andersonjeccel committed:
removing blank lines from file

andersonjeccel committed:
more styling

andersonjeccel committed:
final styling

andersonjeccel committed:
improving js logic

andersonjeccel committed:
polishing a bit more

andersonjeccel committed:
small fix to hr width

andersonjeccel committed:
fix: delete emails routeBase (#14335)

andersonjeccel committed:
fix: theme upload width (#14334)

andersonjeccel committed:
fix: disabled row hardcoded bg color (#14333)

andersonjeccel committed:
fix @matbcvo

matbcvo committed:
Fix "This form should not contain extra fields" error in campaign event form (#14325)

* add allow_extra_fields

* move PRE_SUBMIT listener under event type condition, remove allow_extra_fields

andersonjeccel committed:
improving search stability when content-overlay is present

escopecz committed:
Fixing the best send hour widget (#14358)

andersonjeccel committed:
fix test @matbcvo

matbcvo committed:
fix functional test for GlobalSearch

andersonjeccel committed:
Apply suggestions from code review

Co-authored-by: Rahul Shinde <shinde.r.a@gmail.com>

andersonjeccel committed:
removing unused code

caiqueportela committed:
Don't track downloads that came from non-requests

caiqueportela committed:
Created tests to ensure that the download tracking functionality does not break when there are no requests

andersonjeccel committed:
Merge branch '5.2' into ui-ux-search-almost-everything

kuzmany committed:
refactor: Fix Twig template syntax for form field rendering

fedys committed:
EventRepository::getContactPendingEvents() considers campaign's publishUp and publishDown fields (#14303)

caiqueportela committed:
Fixed formatting, removed unnecessary check, and adjusted check during test execution.

escopecz committed:
Merge pull request #14353 from andersonjeccel/ui-ux-search-almost-everything

* gs styles

* wip

* regression: restoring list groups

* improving focus state

* improving height

* styling for assets based on file type

* improving search item styling

* ok, assets (final)

* expanding publishstatus badge to support simplified mode

* applying badge to asset

* up font-size

* asset again

* improving contact results

* improving landing page results

* improving user results

* improving results: campaigns

* improving results: emails

* improving results: forms

* improving results: pages

* improving results: point actions

* improving results: point triggers

* improving results: reports

* improving results: stages

* improving results: roles

* global search styles improvement

* JS for gsearch

* final adjustments to the template

* removing blank lines from file

* more styling

* final styling

* improving js logic

* polishing a bit more

* small fix to hr width

* improving search stability when content-overlay is present

* fix test @matbcvo

* fix functional test for GlobalSearch

* Apply suggestions from code review

Co-authored-by: Rahul Shinde <shinde.r.a@gmail.com>

* removing unused code

---------

Co-authored-by: Martin Vooremäe <martin.vooremae@gmail.com>
Co-authored-by: Rahul Shinde <shinde.r.a@gmail.com>

escopecz committed:
Merge pull request #14337 from andersonjeccel/bug-upcoming-emails-layout

fix: upcoming emails layout

escopecz committed:
Merge pull request #14258 from patrykgruszka/DPMMA-2936_grapesjs-rte-initializer

DPMMA-2936 grapesjs ckeditor custom initializer

shinde-rahul committed:
Merge branch '5.x' into block-access-to-contact-information-updated

escopecz committed:
Merge pull request #14312 from escopecz/campaign-event-same-day-fix

Campaign event: same day execution fix

escopecz committed:
Merge pull request #14360 from caiqueportela/fix-cron-asset-track

Do not track attachment downloads when there is no request

escopecz committed:
Merge pull request #14367 from biozshock/saml-test

SAMl login fix and test.

shinde-rahul committed:
Merge branch '5.2' into block-access-to-contact-information-updated

escopecz committed:
Merge pull request #14277 from shinde-rahul/log-hashed-sensitive-data

Plugins are not logging the security token

escopecz committed:
Merge pull request #14216 from shinde-rahul/14105-import-notification-for-failed

Notification message to include the reason

escopecz committed:
Merge pull request #14361 from kuzmany/fix-integration-issue

Fix integration variable twig issue

escopecz committed:
Merge pull request #2437 from acquia/MAUT-12062

Fixing the audit log widget when a contact is deleted

escopecz committed:
Increasing Mautic version to 5.2.1

escopecz committed:
Merge pull request #14389 from escopecz/bumping-5.2.1

Increasing Mautic version to 5.2.1

JonasLudwig1998 committed:
Fixing bug in recent activity widget

andersonjeccel committed:
check if input exists before proceeding

JonasLudwig1998 committed:
Merge branch '5.2' into fix-recent-activity-widget

Hugo-Prossaird committed:
style: update decision/action panel colors

Hugo-Prossaird committed:
feature: use Helper Button in segment's filters

driskell committed:
chore: Resolve paste event showing in campaign builder when nothing to paste

escopecz committed:
Merge pull request #14404 from Hugo-Prossaird/update-color-CampaignEventPanel

Update decision/action panel colors in campaign's builder

JonasLudwig1998 committed:
Fixing marketplace table so that select all works

laurielim committed:
Rename function to be more descriptive

laurielim committed:
Toggle "No" radio button before triggering change in Mautic.toggleYesNo

laurielim committed:
Fix condition for showAnonymizeWarningMessage

escopecz committed:
Merge pull request #14407 from driskell/patch-3

[Campaign] In the builder when adding an event in 5.2, it always shows the paste option even if nothing to paste

Hugo-Prossaird committed:
fix: add ghost class on inactive button, primary class on active button

JonasLudwig1998 committed:
Merge branch '5.2' into fix-marketplace-select-all

kuzmany committed:
fix: Correct campaign unpublished translation key in notification subject

JonasLudwig1998 committed:
Fixing problem with single search operators in search fields

JonasLudwig1998 committed:
Making sure operators are only treated as operators when written in uppercase

escopecz committed:
Merge pull request #14413 from JonasLudwig1998/fix-marketplace-select-all

Fixing marketplace table so that selecting all works

andersonjeccel committed:
Merge branch '5.2' into fix-quick-filters-js-console-error

Hugo-Prossaird committed:
refactoring: Use utility class

Co-authored-by: Anderson J. Eccel <116097999+andersonjeccel@users.noreply.github.com>

Hugo-Prossaird committed:
refactoring: Remove duplicate css (set by utility class)

Co-authored-by: Anderson J. Eccel <116097999+andersonjeccel@users.noreply.github.com>

Hugo-Prossaird committed:
fix: remove unexpected margin

escopecz committed:
Merge pull request #14421 from kuzmany/fix-missing-campaign-notificatin-title

FIx translation for campaign unpublished notification

mzagmajster committed:
Wrap renderView method to handle tpl overwrites

eventAwareRenderView: We added this method, because ajax call needs to dispatch event to give plugins chance to replace templates.

driskell committed:
fix: The VIEW_INJECT_CUSTOM_CONTENT event was not getting dispatched with a valid viewName during AJAX content updates

driskell committed:
fix: Fix broken layout of My Categories on preference page (make them match public segments list)

driskell committed:
fix: Fix category selection on admin preference panel and segment filter only showing first 10

driskell committed:
fix: Doctrine migrations no longer result in No Active Transaction errors after DDL statements when using PHP 8

escopecz committed:
Merge pull request #14405 from Hugo-Prossaird/udpate-segment-filter-buttons

UI: update segment filter icons theme

escopecz committed:
Merge pull request #14420 from Hugo-Prossaird/fix-anonymous-contact-filter-style

Fix style anonymous filter in contact list

andersonjeccel committed:
Fixing aria-checked location in the HTML structure

andersonjeccel committed:
Merge branch '5.2' into bugfix-conditional-fields-not-toggling

biozshock committed:
Add missing "isIndexed" and "charLegthLimit" fields to the API response of Contact Fields.

escopecz committed:
Merge pull request #14416 from laurielim/bugfix-conditional-fields-not-toggling

[UI] Fix conditional toggle state not changing

escopecz committed:
Merge pull request #14442 from biozshock/fix-14364

Add missing "isIndexed" and "charLegthLimit" fields to the API response of Contact Fields.

JonasLudwig1998 committed:
Getting rid of issues with defined but empty log.route

escopecz committed:
Resolving a 500 error on asset that was not found

escopecz committed:
PHPSTAN: Asset must exist

escopecz committed:
Merge pull request #14439 from driskell/fix-category-dropdown

fix: Fix category selection on admin preference panel and segment filter only showing first 10

JonasLudwig1998 committed:
Fix SearchStringHelper.php so that tests in CI workflow won't fail

escopecz committed:
Merge pull request #14415 from JonasLudwig1998/fix-single-search-operators

Fixing problem with single search operators in search fields

levente999 committed:
Point Triggers broken: Triggers vanish after first invocation #13817

JonasLudwig1998 committed:
Implementing change request in recentactivity.html.twig

JonasLudwig1998 committed:
Implementing change request in recentactivity.html.twig

JonasLudwig1998 committed:
Implementing change request in recentactivity.html.twig

escopecz committed:
Merge pull request #14434 from mzagmajster/mcp738_mautic_tpl_ajax_rendering_fix

Fix Not allowing template rewrites on ajax page load

LordRembo committed:
add allowed protocols for links in CK5, so people can add phone links (and some others)

LordRembo committed:
Update grapesjs (minor version) + grapesjs-mjml (bugfixes)

escopecz committed:
Merge pull request #14454 from LordRembo/update-grapesjs-20250113

Update GrapesJS to latest version

escopecz committed:
Merge pull request #14453 from LordRembo/fix-14197-link-protocols

Add allowed protocols for links in CK5, so people can add phone links

escopecz committed:
Merge pull request #14396 from JonasLudwig1998/fix-recent-activity-widget

Fixing bug in recent activity widget

patrykgruszka committed:
fix: [DPMMA-3039] Company lookup limit

fix: [DPMMA-3039] change default limit and fix phpstan

fix: [DPMMA-3039] update company list limit

fix: [DPMMA-3039] refactor EntityLookupChoiceLoader::fetchChoices

fix: [DPMMA-3039] test testMax100CompaniesShouldBeFetchedOnContactEditAction

JonasLudwig1998 committed:
Change info about cached data timeout

Hugo-Prossaird committed:
fix: allow preview email by contact for non-admin user

kuzmany committed:
Merge branch '5.2' into fix-email-preview-by-non-admin

escopecz committed:
Merge pull request #14448 from levente999/bugfix-13817

Point Triggers broken: Triggers vanish after first invocation #13817

fedys committed:
Fix FormSubscriberTest

escopecz committed:
Merge pull request #14474 from fedys/fix-form-subscriber-test-5.2

Fix FormSubscriberTest

escopecz committed:
Merge branch '5.2' into fix-email-preview-by-non-admin

escopecz committed:
Merge pull request #14440 from driskell/fix-no-active-transaction

fix: Doctrine migrations no longer result in No Active Transaction errors after DDL statements when using PHP 8

escopecz committed:
Merge pull request #14397 from andersonjeccel/fix-quick-filters-js-console-error

[UI] Popovers stopped working on pages without search field

escopecz committed:
Merge pull request #14471 from Hugo-Prossaird/fix-email-preview-by-non-admin

fix: allow preview email by contact for non-admin user

driskell committed:
fix: Move @defaults theme contents to assets and CoreBundle to ensure accessible in composer installation

patrykgruszka committed:
Merge branch '5.2' into DPMMA-3039_company-lookup-limit-gh

patrykgruszka committed:
Merge branch '5.2' into DPMMA-2957_message-queue-fix

driskell committed:
fix: Fix upsert sometimes storing incorrect ID in entity from a previous upsert operation

driskell committed:
fix: Creating or updating a contact via the Rest API discards seconds for date time fields

RCheesley committed:
Merge branch '5.2' into m5-cached-data-timeout

escopecz committed:
Merge pull request #14425 from driskell/fix-missing-defaults

fix: Fix missing @defaults theme folder in composer installation

escopecz committed:
Merge pull request #14484 from driskell/fix-datetime-api-seconds

fix: Creating or updating a contact via the Rest API discards seconds for date time fields

escopecz committed:
Merge pull request #14438 from driskell/fix-preference-display

fix: Fix broken layout of My Categories on preference page (make them match public segments list)

escopecz committed:
Merge pull request #14482 from driskell/fix-upsert-on-duplicate-key-wrong-id

fix: Fix upsert sometimes storing incorrect ID in entity from a previous upsert operation

escopecz committed:
Merge pull request #14437 from driskell/fix-missing-viewname

fix: The VIEW_INJECT_CUSTOM_CONTENT event was not getting dispatched with a valid viewName during AJAX content updates

bastolen committed:
Fix template block so that the campaign editor has the correct format again

bastolen committed:
Fix template block so that the campaign editor has the correct format again

bastolen committed:
Merge branch 'template-fix' of github.com:bastolen/mautic into template-fix

escopecz committed:
Merge pull request #14491 from bastolen/template-fix

Fix template for Campaign Editor

matbcvo committed:
use composer command

matbcvo committed:
fix e2e db populator

matbcvo committed:
add APP_ENV to e2e-test composer alias

matbcvo committed:
add e2e-tests workflow from the 6.x branch

matbcvo committed:
refactor command for readability

escopecz committed:
Merge pull request #14493 from matbcvo/e2e-backport-5.2

Fix failing E2E test

patrykgruszka committed:
fix: [DPMMA-3048] campaign change membership lead detach

escopecz committed:
Merge pull request #14315 from shinde-rahul/block-access-to-contact-information-updated

Block access to contact information updated

patrykgruszka committed:
fix: [DPMMA-3048] revert changes in AbstractCampaignCommand

patrykgruszka committed:
fix: [DPMMA-3048] phpstan

escopecz committed:
Merge pull request #14497 from patrykgruszka/DPMMA-3048_fix-lead-detach-in-change-campaign-action-gh

DPMMA-3048 Fix campaign execution stuck due to incorrect lead detachment in membership change action

andersonjeccel committed:
fix: Confirmation text being escaped incorrectly

escopecz committed:
Merge pull request #14508 from andersonjeccel/fix-translations-confirmation-text

fix: Confirmation text being escaped incorrectly

escopecz committed:
Fixing the right menu height on smaller screens

escopecz committed:
Merge pull request #14517 from escopecz/right-menu-viewport-fix-5

Right menu viewport fix

RCheesley committed:
Bump version to 5.2.2 in readiness for release (#14518)

driskell committed:
fix: Email preview now works again even if unpublished or expired

driskell committed:
fix: Fix font selection in CKEditor not including fallback fonts in output

patrykgruszka committed:
fix: [DPMMA-3039] update the limit condition

escopecz committed:
The fix is not present in the community. The test is still valualbe

escopecz committed:
The audit log for deleted contacts was somehow missing (?) so I added it to make the tests successful

escopecz committed:
Nope, reverting. I was looking into bad subscriber

escopecz committed:
The widget height must be higher to fit in more than 2 rows :facepalm:

escopecz committed:
Merge pull request #14541 from escopecz/audit-log-on-deleted

Fixing the audit log widget when a contact is deleted

escopecz committed:
Merge pull request #14549 from escopecz/segment-default-timezone

Fixing segment building with default timezone

escopecz committed:
Merge pull request #14540 from escopecz/php-warning-fix-tracking

Email click tracking fix, PHP warning fix

patrykgruszka committed:
fix: [DPMMA-3031] Improve email address handling for non-ASCII names

patrykgruszka committed:
fix: [DPMMA-3031] use symfony endoder to get the address length

patrykgruszka committed:
fix: [DPMMA-3031] mail helper unit test

Hugo-Prossaird committed:
fix: remove z-index from focus preview to avoid issue with placeholder

Hugo-Prossaird committed:
fix: too much padding for select input with form-control class

escopecz committed:
Merge pull request #14539 from driskell/fix-ckeditor-font

fix: Fix font selection in CKEditor not including fallback fonts in output

patrykgruszka committed:
fix: [DPMMA-3031] address length limit config

abhisekmazumdar committed:
Add testcase.

abhisekmazumdar committed:
Fix cs.

Commit:
Check permission on original entity for cloning.

escopecz committed:
Merge pull request #14577 from patrykgruszka/DPMMA-3031_address-length-limit-gh

DPMMA-3031 Configurable email address length limit to prevent delivery issues

escopecz committed:
Merge pull request #14525 from driskell/fix-preview-require-published

fix: Email preview now works again even if unpublished or expired

abhisekmazumdar committed:
Add functional test.

abhisekmazumdar committed:
Fix cs.

andersonjeccel committed:
[UI] Move Help to top navbar-right

escopecz committed:
Merge pull request #14580 from mallezie/fix-email-clone-permission

Check permission on original entity for email cloning.

escopecz committed:
Merge pull request #14568 from Hugo-Prossaird/fix-focus-preview-placeholder

Fix Focus preview with placeholder

escopecz committed:
Merge pull request #14569 from Hugo-Prossaird/fix-widget-select-padding

fix: too much padding for select input with form-control class

abhisekmazumdar committed:
Replace assertTrue with assertResponseIsSuccessful.

abhisekmazumdar committed:
Merge branch '5.2' into fix-group-label-sync

escopecz committed:
Merge pull request #14582 from andersonjeccel/ui-move-help-to-top-navbar

[UI] Move Help to top navbar-right

escopecz committed:
Merge pull request #14560 from abhisekmazumdar/fix-group-label-sync

FIX: Ignoring of custom option list for mapped fields (#14117)

lenonleite committed:
- add validation by extension and mimetype

lenonleite committed:
- add validation by extension and mimetype

lenonleite committed:
- add translations
- update config file

Hugo-Prossaird committed:
fix: duplicate title on campaign source change + adapt style

nick-vanpraet committed:
Merge pull request #14247 from patrykgruszka/DPMMA-2957_message-queue-fix

DPMMA-2957 Prevent ORM error when sending multiple messages to one Lead

rishithreddy89 committed:
Update point.css

RCheesley committed:
Merge branch '5.2' into issue-14412-rr

VishnuGurrapu committed:
fixed issue-#14559

RCheesley committed:
Merge branch '5.2' into fix-duplicate-title-campaign-source

nick-vanpraet committed:
Merge pull request #14617 from rishithreddy89/issue-14412-rr

Fixed: Delete Events for Point Triggers are weird #14412

nick-vanpraet committed:
Merge pull request #14615 from Hugo-Prossaird/fix-duplicate-title-campaign-source

Duplicate title on campaign source change

VishnuGurrapu committed:
Update app.css

nick-vanpraet committed:
Increasing version to 5.2.3

laurielim committed:
Ensure that the data-theme value is retrieved if nested span is clicked instead of the element holding the attribute.

laurielim committed:
Remove console logs

nick-vanpraet committed:
Merge pull request from GHSA-73gx-x7r9-77x2

nick-vanpraet committed:
Merge pull request from GHSA-8xv7-g2q3-fqgc

nick-vanpraet committed:
Merge pull request from GHSA-4w2w-36vm-c8hf

nick-vanpraet committed:
Merge pull request #14634 from nick-vanpraet/bump-5.2.3

Increasing version to 5.2.3

driskell committed:
fix: Fix contact preview lookup on email not performing a lookup

escopecz committed:
Merge pull request #14092 from johbuch/5-x-fix-13570-incorrect-banner-theme-deletion

fix #13570 - incorrect banner when multiple theme deletion

escopecz committed:
Merge pull request #14638 from laurielim/issue-14338-custom-html-content-hidden

Fix issue #14338 Custom HTML Content hidden when creating email in Code Mode

escopecz committed:
Merge pull request #14619 from VishnuGurrapu/issue-14559

fixed #14559 :When deleting a theme the confirmation modal is not readable

escopecz committed:
Merge pull request #14648 from driskell/fix-contact-preview-lookup

fix: Fix contact preview lookup on email not performing a lookup

kuzmany committed:
refactor: Remove unnecessary whitespace and hyphens in Twig template

escopecz committed:
Merge pull request #14461 from patrykgruszka/DPMMA-3039_company-lookup-limit-gh

DPMMA-3039 Company lookup limit

escopecz committed:
Merge pull request #14677 from kuzmany/fix-campaign-toogle

Fix space between toggle and campaign name

npracht committed:
fix wording and encoding issue

JonasLudwig1998 committed:
Fixing disable cache so that 0 works

JonasLudwig1998 committed:
Merge remote-tracking branch 'origin/m5-cached-data-timeout' into m5-cached-data-timeout

npracht committed:
Fix integration segment filter

escopecz committed:
Merge pull request #14663 from escopecz/asset-404-fix

Asset 404 fix

Hugo-Prossaird committed:
refactoring: simplify notification header rendering

escopecz committed:
Merge pull request #14711 from mautic/fix/notification-encoding-and-wording

fix wording and encoding issue in notifications

escopecz committed:
Merge pull request #14467 from JonasLudwig1998/m5-cached-data-timeout

Fix cached data timeout logic in m5

escopecz committed:
Merge pull request #14712 from npracht/fix/salesforce-segment-filter

Salesforce campaign segment filter select fixed

lenonleite committed:
- add validation to don't anonymize values less than 64 characters.

lenonleite committed:
Merge remote-tracking branch 'origin/5.2' into feature/add-event-anonymize

# Conflicts:
#	app/bundles/LeadBundle/Translations/en_US/messages.ini

lenonleite committed:
Merge remote-tracking branch 'origin/5.2' into feature/add-event-anonymize

lenonleite committed:
- improve Anonymize and tests

lenonleite commented:
This PR is obsolete because it introduces a new feature that will not be included in Mautic 5. A new PR with the same feature has been opened for Mautic 7.
